### ISSUE TITLE
Link Error: using dlib library with cuda 

### ISSUE THREAD

**Issue by grzlog** [davisking/dlib#1294]
I tried to build my own program to do face detection with dlib and cuda. 
--Using Win10, VS2015
--CMAKE version = 3.11.1
--CUDA=9.0 CUDNN=7
--Dlib = v19.10
--main idea based on [http://dlib.net/dnn_mmod_face_detection_ex.cpp.html](url)
Following is what I did to compile dlib first:
`mkdir build; cd build; `
`cmake -G "Visual Studio 14 2015 Win64" -DCMAKE_INSTALL_PREFIX=install .. `
`cmake --build . --config Release --target INSTALL`

so far so good, I passed cuda and cudnn test, so I enabled cuda. In my programme, I use find_package(dlib). The problem is when I compiled my program, it turns out this **error**:

`  dlib19.10.0_release_64bit_msvc1900.lib(dlib_generated_cuda_dlib.cu.obj) : error LNK2019: 无法解析的外部符号 USER_ERROR__incons
istent_build_configuration__see_dlib_faq_2，该符号在函数 "void __cdecl dynamic initializer for 'dlib_check_not_configured_helper_variable''(void)" (??__Edlib_check_not_configured_helper_variable@@YAXXZ)`
`dlib19.10.0_release_64bit_msvc1900.lib(dlib_generated_cusolver_dlibapi.cu.obj) : error LNK2001: 无法解析的外部符号 USER_ERROR__inconsistent_build_configuration__see_dlib_faq_2`

btw: if I set `DLIB_USE_CUDA=OFF` when compile dlib, then my program can be compiled correctly.

And if I use the way you recommended (cuda enabled), 'add_subdirectory(../dlib)' and put my source code into examples folder, then compiling succeed as well. 

Could you please share some ideas? Thanks in advance!

**Comment by davisking** [comment]
>  USER_ERROR__incons istent_build_configuration__see_dlib_faq_2

Did you read FAQ 2?

**Comment by grzlog** [comment]
Yes, I read it. But I don't very understand it. If I want to use dlib separately, what should I do?

**Comment by davisking** [comment]
You read this? http://dlib.net/faq.html#WhydoIgetUSERERRORinconsistentbuildconfigurationseedlibfaq2

I don't know how to explain it more simply.  Don't manually copy files around. You are messing it up.

**Comment by grzlog** [comment]
Thanks for quick response. 

Does `Don't manually copy files around.` mean `-DCMAKE_INSTALL_PREFIX=install` is wrong?

**Comment by davisking** [comment]
No, that's fine.

If you want help you need to post complete instructions, that someone else
can follow, to reproduce your problem.  Your post as is is way too vague.
You basically say, "I do it right and it doesn't work".  But that's not
right, otherwise it would have worked. Post something someone can run
exactly and get your result.

**Comment by grzlog** [comment]
ok, thanks so much! Maybe I shall make it simple:
1. I downloaded dlib v19.10
2. then compiled dlib as follow:
`cd dlib-19.10;   cd dlib;   mkdir build;   cd build`
`cmake -G "Visual Studio 14 2015 Win64" -T host=64 -DCMAKE_INSTALL_PREFIX=install ..`
`cmake --build . --config Release --target INSTALL`
so there would be an "install" folder which contains lib and include.
-----------------------------
my program now have cmakelist.txt:

`cmake_minimum_required(VERSION 3.2)`
`project(testdlib)`
`set(CMAKE_CXX_STANDARD 11)`

`find_package(dlib REQUIRED)`
`set(SOURCE_FILES src/dnn_mmod_face_detection_ex.cpp)`
`add_executable(${PROJECT_NAME} ${SOURCE_FILES})`
`target_include_directories(${PROJECT_NAME} PUBLIC dlib::dlib)`
`target_link_libraries(${PROJECT_NAME} PUBLIC dlib::dlib)`

and my src file is just the example [http://dlib.net/dnn_mmod_face_detection_ex.cpp.html](url)


then I tried to compile my own program with:
`mkdir build;   cd build`
`cmake -G "Visual Studio 14 2015 Win64" ..`
`cmake --build . --config Release`

then the errors just appear as I showed above.

Thanks so much! Does anyone have any idea?

**Comment by davisking** [comment]
Hmm, this seems to be an issue with cmake not propagating compiler settings
to nvcc consistently.  I just pushed a change to github that should fix
this.  Give the latest code a try and let me know if it works now.

**Comment by grzlog** [comment]
cool! It works now.  Thank you~ appreciate

**Comment by davisking** [comment]
No problem, thanks for reporting this.
